pkg/promutil/migrate.go

```diff
@@ -285,13 +295,11 @@ func EnsureLabelConsistencyAndRemoveDuplicates(metrics []*PrometheusMetric, obse
 	output := make([]*PrometheusMetric, 0, len(metrics))
 
 	for _, metric := range metrics {
-		for observedLabels := range observedMetricLabels[metric.Name] {
-			if _, ok := metric.Labels[observedLabels]; !ok {
-				metric.Labels[observedLabels] = ""
-			}
+		for observedLabels := range observedMetricLabels[metric.Name()] {
+			metric.AddIfMissingLabelPair(observedLabels, "")
```

Comment:
This part of the change might be interesting to benchmark because it's going to be called a lot. I don't think the number of labels is going to be all that large in most cases would be interesting to see the CPU impact for slice scanning vs map lookup on the various use cases.